.github/workflows/CoverageCollection.yml

```diff
@@ -67,30 +97,31 @@ jobs:
       - name: Collect coverage
         continue-on-error: true
         run: |
-          python -m pytest -rA --cov=.. --cov-config=tests/.coveragerc tests/unit --color=yes
+          [ 'x${{ inputs.coverage_config }}' != 'x' ] && PYCOV_ARGS='--cov-config=${{ inputs.coverage_config }}' || unset PYCOV_ARGS
+          python -m pytest -rA --cov=.. $PYCOV_ARGS ${{ inputs.unittest_directory }} --color=yes
 
       - name: Convert to cobertura format
         run: coverage xml
 
       - name: Convert to HTML format
         run: |
           coverage html
-          rm htmlcov/.gitignore
+          rm ${{ steps.getVariables.outputs.coverage_report_html_directory }}/.gitignore
```

Comment:
I think that the content is flatenned in the zipfile. Hence, the path used here should have no effect when downloading the artifact.

Reply:
ok. In GitLab the path is preserved, which ensures the artifact is recreated in another job at the same location :)
(which I very like - that's the purpose of artifacts - job-to-job data transfer ...).

Reply:
In GitHub, job-to-job data transfer is just one of the usages of artifacts. Hence, there is the `path` field when downloading the artifact, in case you want to customise the location. That is actually useful in several cases, because it avoids an step to move the content (either before or after). Typical case is producing multiple HTML artifacts and then downloading them in a given hierarchy before pushing to GitHub Pages: https://github.com/pyTooling/Actions/blob/main/.github/workflows/PublishToGitHubPages.yml#L53-L71